flow/e2e/clickhouse/peer_flow_ch_test.go

```diff
@@ -1029,3 +1030,148 @@ func (s ClickHouseSuite) Test_Nullable_Schema_Change() {
 	env.Cancel(s.t.Context())
 	e2e.RequireEnvCanceled(s.t, env)
 }
+
+// Test_Normalize_Metadata_With_Retry tests the chunking normalization
+// with a push to ClickHouse thrown in via renaming a target table.
+func (s ClickHouseSuite) Test_Normalize_Metadata_With_Retry() {
+	if _, ok := s.source.(*e2e.MySqlSource); ok {
```

Comment:
You can cast this to `PostgresSource` & get direct access to PostgresConnector that way, which removes needing `Query` method on interface